### PR TITLE
Windows k8s support changes, merge flex volume driver windows changes…

### PR DESCRIPTION
**Pull request by vidhutsingh**
Created a new pull request to support k8s on Windows platform .

### CODE REVIEW

**Comment by vidhutsingh** [comment]
I have reseted my fork and created a new pull request.

File: pkg/flexvol/flexvol.go [hunk 1]
```diff
@@ -432,6 +396,9 @@ func retryGetDockerPathAndMetadata(flexvolPath, devPath string) (string, string,
 			}
 			return dockerPath, metadata, err
 		}
+		if err != nil {
```
Comment (shivamerla) [inline pkg/flexvol/flexvol.go]: unreachable code..did it compile for you?

File: pkg/flexvol/flexvol.go [hunk 2]
```diff
@@ -348,10 +307,15 @@ func retryGetVolumeNameFromMountPath(k8sPath, dockerPath string) (string, error)
 
 func getMountID(path string) (string, error) {
 	log.Tracef("getMountID called with %v\n", path)
+	// Replace backward slash to forward slash.
```
Comment (shivamerla) [inline pkg/flexvol/flexvol.go:310]: I think the caller has to take care of this(or if caller is windows specific, then no translation is necessary)..

File: pkg/flexvol/flexvol_linux.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	"fmt"
+	"github.com/hpe-storage/common-host-libs/docker/dockervol"
+	"github.com/hpe-storage/common-host-libs/linux"
+	log "github.com/hpe-storage/common-host-libs/logger"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+// TO-DO - We should use CHAPI module instead of using PowerShell command
```
Comment (shivamerla) [inline pkg/flexvol/flexvol_linux.go]: This applies to other file..

**Comment by shivamerla** [comment]
@vidhutsingh added some comments, when you fix them also run "git commit --amend --signoff" to pass DCO checks..

File: pkg/flexvol/flexvol.go (hunk 1 above)
Comment (vidhutsingh) [reply pkg/flexvol/flexvol.go]: Yes this has compiled and possiblly due to multiple merge it cropped in . thx for catching.

File: pkg/flexvol/flexvol.go (hunk 2 above)
Comment (vidhutsingh) [reply pkg/flexvol/flexvol.go:310]: getMountID is called at multiple places, it is common for linux and windows. We are simply extracing mountID from the path here and it is handled for both windows and linux . So this looks clean .

**Review by shivamerla — APPROVED** [review]

**Pull request merged**